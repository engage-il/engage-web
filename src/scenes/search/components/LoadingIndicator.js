// @flow
import React, { Component } from 'react'
import { css } from 'glamor'
import { BillAnimation, billRule } from './BillAnimation'
import { stylesheet, colors, mixins } from 'shared/styles'

export class LoadingIndicator extends Component {
  props: {
    isLoading: boolean
  }

  // lifecycle
  render () {
    return <BillAnimation disableAppear>
      {this.renderIndicator()}
    </BillAnimation>
  }

  renderIndicator (): ?React$Element<*> {
    if (!this.props.isLoading) {
      return null
    }

    return <div key='indicator' {...css(rules.container, billRule)}>
      <div {...rules.topBar} />
      <div {...rules.bottomBar} />
    </div>
  }
}

function animation (start: string, end: string, duration: number): Object {
  return {
    animationName: css.keyframes({ from: { width: start }, to: { width: end } }),
    animationDuration: `${duration}s`,
    animationDirection: 'alternate',
    animationIterationCount: 'infinite'
  }
}

const bar = {
  height: 22,
  borderRadius: 3,
  backgroundColor: colors.neutralShadow
}

const rules = stylesheet({
  container: {
    ...mixins.shadows.low(),
    ...mixins.borders.low(),
    height: 49,
    padding: 15,
    marginBottom: 15,
    backgroundColor: colors.neutral
  },
  topBar: {
    ...bar,
    ...animation('10%', '13%', 1.5),
    marginBottom: 5
  },
  bottomBar: {
    ...bar,
    ...animation('8%', '15%', 1.2)
  }
})
