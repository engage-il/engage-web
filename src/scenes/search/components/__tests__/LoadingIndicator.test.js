/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { LoadingIndicator } from '../LoadingIndicator'

let subject
let props

function loadSubject () {
  subject = shallow(<LoadingIndicator {...props} />)
}

function test () {
  expect(subject).toMatchSnapshot()
}

beforeEach(() => {
  props = {
    isLoading: false
  }
  subject = null
})

describe('#render', () => {
  it('when loading', () => {
    props.isLoading = true
    loadSubject()
    test()
  })

  it('when not loading', () => {
    loadSubject()
    test()
  })
})
