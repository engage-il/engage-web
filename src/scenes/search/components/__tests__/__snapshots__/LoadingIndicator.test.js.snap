// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render when loading 1`] = `
<BillAnimation
  disableAppear={true}
>
  <div
    data-css-1judezb=""
  >
    <div
      data-css-1f89q0c=""
    />
    <div
      data-css-15mi9ej=""
    />
  </div>
</BillAnimation>
`;

exports[`#render when not loading 1`] = `
<BillAnimation
  disableAppear={true}
/>
`;
